src/dotnet/commands/dotnet-build3/MSBuildForwardingApp.cs

```diff
@@ -0,0 +1,61 @@
+﻿using Microsoft.DotNet.Cli.Utils;
+using Microsoft.DotNet.InternalAbstractions;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Microsoft.DotNet.Cli
+{
+    public class MSBuildForwardingApp : ForwardingApp
```

Comment:
Please, use composition and not inheritance here.

Your MSBuildForwardingApp can simply have a ForwardingApp inside of it, instead of inheriting from it. You are really gaining nothing from the inheritance but the Execute(), which in the composition case would just be forwardingApp.Execute().